crates/wasm3-runtime/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+// FIXME: there's a lot of `.map_err(|e| anyhow!("{}", e))` in here, because
```

Comment:
You could use an extension trait to help you here:

```rust
trait Wasm3ResultExt<T> {
  fn to_anyhow(self) -> Result<T, anyhow::Error>;
}

impl<T> Wasm3ResultExt<T> for Result<T, Wasm3Error> {
  fn to_anyhow(self) -> Result<T, anyhow::Error> { self.map_err(|e| anyhow!("{}", e)) }
}
```

And it'd be used like

```rust
let env = Environment::new().to_anyhow()?;
```

Reply:
I've implemented this, and also worked around the soundness bug in wasm3